tmt/base.py

```diff
@@ -830,15 +837,24 @@ def __init__(
         self.finish = tmt.steps.finish.Finish(
             plan=self, data=self.node.get('finish'))
 
-        # Test execution context defined in the plan
-        self._plan_context = self.node.get('context', dict())
-
         self._update_metadata()
 
     def _expand_node_data(self, data):
         """ Recursively expand variables in node data """
         if isinstance(data, str):
-            return os.path.expandvars(data)
+            # expand env variables
+            data_expanded_vars = os.path.expandvars(data)
+            # convert context values to string and define env variables using context
+            # keys and values
+            list_context_string = [
+                (key, ','.join(value)) if isinstance(value, list) else (key, str(value))
```

Comment:
@happz With normalization in place, is this test and else block still necessary?

Reply:
I don't think so, context should be always `Dict[str, List[str]]`, even if there's just a single value in the list.

Reply:
OK. I have tried to remove it.

Reply:
One more thing, `modify_environ()` takes `EnvironmentType` which is a fancy name for `Dict[str, str]`, therefore instead of a list you need to prepare a key/value mapping.

Type annotations of `base.py` are under review, no mypy warning so far :/

Reply:
@happz Updated. Is it correct now?

Reply:
Looks correct to me.